pageserver/src/metrics.rs

```diff
@@ -94,15 +94,37 @@ pub(crate) static READ_NUM_FS_LAYERS: Lazy<Histogram> = Lazy::new(|| {
 });
 
 // Metrics collected on operations on the storage repository.
-pub(crate) static RECONSTRUCT_TIME: Lazy<Histogram> = Lazy::new(|| {
-    register_histogram!(
-        "pageserver_getpage_reconstruct_seconds",
-        "Time spent in reconstruct_value (reconstruct a page from deltas)",
-        CRITICAL_OP_BUCKETS.into(),
-    )
-    .expect("failed to define a metric")
+
+pub(crate) struct ReconstructTimeMetrics {
+    ok: Histogram,
+    err: Histogram,
+}
+
+pub(crate) static RECONSTRUCT_TIME: Lazy<ReconstructTimeMetrics> = Lazy::new(|| {
+    static IMPL: Lazy<HistogramVec> = Lazy::new(|| {
+        register_histogram_vec!(
+            "pageserver_getpage_reconstruct_seconds",
+            "Time spent in reconstruct_value (reconstruct a page from deltas)",
+            &["result"],
+            CRITICAL_OP_BUCKETS.into(),
+        )
+        .expect("failed to define a metric")
+    });
```

Comment:
I think this will be accessed "right away", and does not need to exist for `static` anymore.

Reply:
Elsewhere: https://github.com/neondatabase/neon/blob/dee2bcca44b63906d0d9361735e6214e44f86325/libs/remote_storage/src/s3_bucket/metrics.rs#L146-L157